src/SonarScanner.MSBuild/BootstrapperClass.cs

```diff
@@ -33,7 +38,13 @@ public class BootstrapperClass
     {
         public const int ErrorCode = 1;
         public const int SuccessCode = 0;
+#if NETCOREAPP2_1
+
+        private const string WarningMessage = "From [Date], we will no longer release new Scanner for .NET versions that target .NET Core 2.1." +
```

Comment:
IMO you can skip having this as a field and move it inside `PostProcess` as `const string` - either way the compiler does the optimization and you'll keep all NETFRAMEWORK logic together.